### ISSUE TITLE
regexp: in url causes problems with mod_sec owasp rules

### ISSUE THREAD

**Issue by nitriques** [symphonycms/symphony-2#2675]
OWASP defines mod_sec rules that prohibits usage of certain SQL keywords (union, regexp, contains).

When filtering entries, I often get a mod_sec hit, which prevents it from running the query.

I think it should be possible to change does words in the UI, but still keep the current words as valid to still be backward compatible, while working with mod_sec OWASP rules

**Comment by michael-e** [comment]
I am not sure how I could help with that. I once tried mod_sec, but I gave up. :-(

**Comment by nitriques** [comment]
>  but I gave up

Yeah most of the people do. But I do run it (I need to), and loosing the ability to filter with 'contains' is rather annoying... Would it be a problem if I replace it with something else (while making sure old url still works) ?

**Comment by michael-e** [comment]
Wouldn't it be the right way to break compatibility? Isn't that mod_sec stuff meant to help _improve_ systems? :-)

I don't know how it should be done, however. Simply translating parameter names wouldn't make it any safer either, IMHO.

**Comment by nitriques** [comment]
> Simply translating parameter names wouldn't make it any safer either, IMHO.

That right. It is more of a 'hosting compatibility' issue. mod_sec can't know for sure that the 'regexp' part is not going to end up in some SQL command. So they blindly forbid that keyword (they also forbid using the word union: a client a mine did not understood why some times she would get redirected to the home page when saving an entry.

The owasp bundle is quite agressive, but I've seen it on a couple of host. They simply to not trust any code and try to be as safe as possible, but yeah it can make apps crash. But in the end, I do feel like my hosting is more "secure" with it enabled.
